presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -475,7 +484,23 @@ public CounterStat getHighMemorySplitSource()
 
                 results.add(new HivePartitionMetadata(hivePartition, Optional.of(partition), partitionSchemaDifference.build(), encryptionInformation));
             }
-
+            StringBuilder warningMessage = new StringBuilder(format("Table '%s' has '%s' out of '%s' partitions unreadable: ", tableName, unreadablePartitionsSkipped, partitionBatch.size()));
+            for (Entry<String, Set<String>> entry : partitionsNotReadable.entrySet()) {
+                if (entry.getValue().size() > 2) {
+                    Iterator it = entry.getValue().iterator();
+                    warningMessage.append("\n" + entry.getKey().toString() + ": " + it.next() + ", " + it.next() + " ...");
+                }
+                else {
+                    warningMessage.append("\n" + entry.getKey().toString() + ": " + entry.getValue().toString());
+                }
+            }
+            if (partitionsNotReadable.entrySet().size() > 2) {
+                if (warningMessage.lastIndexOf("\n") >= 0) {
+                    warningMessage.delete(warningMessage.lastIndexOf("\n"), warningMessage.length());
+                    warningMessage.append("\n...");
+                }
+            }
```

Comment:
We really don't need to be such complicated. Also, there are many bugs in this code block like (1) this should be guarded by an `if`, (2) `.toString()` is redundant, (3) `entry.getValue().toString()` is going to get you the memory pointer, etc. Let's simplify this logic:
```java

if (unreadablePartitionsSkipped > 0) {
    StringBuilder warningMessage = new StringBuilder(format("Table '%s' has '%s' out of '%s' partitions unreadable: ", tableName, unreadablePartitionsSkipped, partitionBatch.size()));
    for (Entry<String, Set<String>> entry : partitionsNotReadable.entrySet()) {
        warningMessage.append(join(", ", entry.getValue()))
                .append("... are due to ")
                .append(entry.getKey())
                .append(". ");
    }
    warningCollector.add(new PrestoWarning(PARTITION_NOT_READABLE, warningMessage.toString()));
}